File: src/m/classes/bamggeom.py
import numpy as np


class bamggeom(object):
    """
    BAMGGEOM class definition

       Usage:
          bamggeom(varargin)
    """

    def __init__(self, *args):  # {{{
        self.Vertices = np.empty((0, 3))
        self.Edges = np.empty((0, 3))
        self.TangentAtEdges = np.empty((0, 4))
        self.Corners = np.empty((0, 1))
        self.RequiredVertices = np.empty((0, 1))
        self.RequiredEdges = np.empty((0, 1))
        self.CrackedEdges = np.empty((0, 0))
        self.SubDomains = np.empty((0, 4))

        if not len(args):
            # if no input arguments, create a default object
            pass

        elif len(args) == 1:
            object = args[0]
            for field in list(object.keys()):
                if field in vars(self):
                    setattr(self, field, object[field])

        else:
            raise TypeError("bamggeom constructor error message: unknown type of constructor call")
    # }}}

    def __repr__(self):  # {{{
        s = "class '%s' object '%s'=\n" % (type(self), 'self')
        s += "    Vertices: %s\n" % str(self.Vertices)
        s += "    Edges: %s\n" % str(self.Edges)
        s += "    TangentAtEdges: %s\n" % str(self.TangentAtEdges)
        s += "    Corners: %s\n" % str(self.Corners)
        s += "    RequiredVertices: %s\n" % str(self.RequiredVertices)
        s += "    RequiredEdges: %s\n" % str(self.RequiredEdges)
        s += "    CrackedEdges: %s\n" % str(self.CrackedEdges)
        s += "    SubDomains: %s\n" % str(self.SubDomains)
        return s
    # }}}

File: src/c/bamg/BamgGeom.cpp
#include "./bamgobjects.h"
#include "../shared/shared.h"

/*Constructors/Destructors*/
BamgGeom::BamgGeom(){/*{{{*/

	this->VerticesSize[0]=0,          this->VerticesSize[1]=0;          this->Vertices=NULL;
	this->EdgesSize[0]=0,             this->EdgesSize[1]=0;             this->Edges=NULL;
	this->TangentAtEdgesSize[0]=0,    this->TangentAtEdgesSize[1]=0;    this->TangentAtEdges=NULL;
	this->CornersSize[0]=0,           this->CornersSize[1]=0;           this->Corners=NULL;
	this->RequiredVerticesSize[0]=0,  this->RequiredVerticesSize[1]=0;  this->RequiredVertices=NULL;
	this->RequiredEdgesSize[0]=0,     this->RequiredEdgesSize[1]=0;     this->RequiredEdges=NULL;
	this->CrackedEdgesSize[0]=0,      this->CrackedEdgesSize[1]=0;      this->CrackedEdges=NULL;
	this->SubDomainsSize[0]=0,        this->SubDomainsSize[1]=0;        this->SubDomains=NULL;

}
/*}}}*/
BamgGeom::~BamgGeom(){/*{{{*/

	xDelete<double>(this->Vertices);
	xDelete<double>(this->Edges);
	xDelete<double>(this->TangentAtEdges);
	xDelete<double>(this->Corners);
	xDelete<double>(this->RequiredVertices);
	xDelete<double>(this->RequiredEdges);
	xDelete<double>(this->CrackedEdges);
	xDelete<double>(this->SubDomains);

}
/*}}}*/
